src/eos_stream.c

```diff
@@ -599,47 +699,68 @@ static void processActionData(txProcessingContext_t *context) {
-        } else if (context->contractName == EOSIO &&
-                   context->contractActionName == EOSIO_SELLRAM) {
-            processEosioSellRam(context);
-        } else if (context->contractName == EOSIO &&
-                   context->contractActionName == EOSIO_UPDATE_AUTH) {
-            processEosioUpdateAuth(context);
-        } else if (context->contractName == EOSIO &&
-                   context->contractActionName == EOSIO_DELETE_AUTH) {
-            processEosioDeleteAuth(context);
-        } else if (context->contractName == EOSIO &&
-                   context->contractActionName == EOSIO_LINK_AUTH) {
-            processEosioLinkAuth(context);
-        } else if (context->contractName == EOSIO &&
-                   context->contractActionName == EOSIO_UNLINK_AUTH) {
-            processEosioUnlinkAuth(context);
+        } else if (context->contractName == EOSIO) {
+            switch (context->contractActionName) {
+            case EOSIO_DELEGATEBW:
+                processEosioDelegate(context);
+                break;
+            case EOSIO_UNDELEGATEBW:
+                processEosioUndelegate(context);
+                break;
+            case EOSIO_REFUND:
+                processEosioRefund(context);
+                break;
+            case EOSIO_VOTEPRODUCER:
+                processEosioVoteProducer(context);
+                break;
+            case EOSIO_BUYRAM:
+            case EOSIO_BUYRAMBYTES:
+                processEosioBuyRam(context);
+                break;
+            case EOSIO_SELLRAM:
+                processEosioSellRam(context);
+                break;
+            case EOSIO_UPDATE_AUTH:
+                processEosioUpdateAuth(context);
+                break;
+            case EOSIO_DELETE_AUTH:
+                processEosioDeleteAuth(context);
+                break;
+            case EOSIO_LINK_AUTH:
+                processEosioLinkAuth(context);
+                break;
+            case EOSIO_UNLINK_AUTH:
+                processEosioUnlinkAuth(context);
+                break;
+            case EOSIO_NEW_ACCOUNT:
+                processEosioNewAccountAction(context);
+                break;
+            default:
+                THROW(EXCEPTION);
+            }
+        }
+        
+        if (++context->currentActionIndex < context->currentActionNumer) {
+            context->state = TLV_ACTION_ACCOUNT;
         } else {
-            THROW(EXCEPTION);
+            context->state = TLV_TX_EXTENSION_LIST_SIZE;
         }
-        context->state = TLV_TX_EXTENSION_LIST_SIZE;
+        
         context->processingField = false;
+        context->actionReady = true;
     }
 }
 
 static parserStatus_e processTxInternal(txProcessingContext_t *context) {
     for(;;) {
+        if (context->confirmProcessing) {
+            context->confirmProcessing = false;
+            return STREAM_CONFIRM_PROCESSING;
+        }
+        if (context->actionReady) {
```

Comment:
This flag is set when an action is ready to be displayed, and user should confirm it.